quantum/process_keycode/process_auto_shift.h

```diff
@@ -22,13 +22,34 @@
 #    define AUTO_SHIFT_TIMEOUT 175
 #endif
 
+#define IS_LT(kc) ((kc) >= QK_LAYER_TAP && (kc) <= QK_LAYER_TAP_MAX)
+#define IS_MT(kc) ((kc) >= QK_MOD_TAP && (kc) <= QK_MOD_TAP_MAX)
+#define IS_RETRO(kc) (IS_MT(kc) || IS_LT(kc))
+#define DO_GET_AUTOSHIFT_TIMEOUT(keycode, record, ...) record
+// clang-format off
+#define get_autoshift_timeout(...) _Generic(        \
+        (DO_GET_AUTOSHIFT_TIMEOUT(__VA_ARGS__, 0)), \
+            keyrecord_t *: (get_autoshift_timeout), \
+            default: get_generic_autoshift_timeout  \
+)(__VA_ARGS__)
+#define AUTO_SHIFT_ALPHA KC_A ... KC_Z
+#define AUTO_SHIFT_NUMERIC KC_1 ... KC_0
+#define AUTO_SHIFT_SPECIAL          \
+             KC_TAB:                \
+        case KC_MINUS ... KC_SLASH: \
+        case KC_NONUS_BSLASH
```

Comment:
This allows `case AUTO_SHIFT_ALPHA:` and the like in the user's `get_autoshift_timeout`. Janky, but I think it should be like this.

Reply:
I don't agree.  This removes a lot of code and obfuscates it, making it less readable/understandable.